### PR TITLE
CAMS-101: pipeline coverage

### PR DESCRIPTION
**Pull request by amorrow-flexion**
# Purpose

To have the build break when line coverage is below 90%

# Major Changes

- Changes to user-interface and backend package.json to add coverage thresholds
- updates to pipeline to run coverage

# Testing/Validation

Tested changes by reducing coverage below 90% on both frontend and backend individually and running builds, verifying that it forces build failure when coverage is below 90% on lines covered.

# Notes

we can add thresholds to the other categories as well

### CODE REVIEW

**Review by governmentSponsored — APPROVED** [review]

**Comment by governmentSponsored** [comment]
@amorrow-flexion the code changes look good. I'm wondering why build failed for backend function if 104/105 tests passed?

**Comment by governmentSponsored** [comment]
fixed merge conflicts by bringing in `main` changes

File: .github/workflows/continuous-deployment.yml [hunk 1]
```diff
@@ -245,8 +245,15 @@ jobs:
...
-        run: CI=true npm test
+      - name: Execute Tests
+        if: ${{ vars.ENVIRONMENT != 'Main-Gov' && github.event_name != 'pull_request' }}
+        run: |
+          CI=true npm run test
+
+      - name: Execute Coverage
+        if: ${{ vars.ENVIRONMENT == 'Main-Gov' || (github.event_name == 'pull_request' && github.base_ref == 'main') }}
+        run: |
+          CI=true npm run coverage
```
Comment (jamesobrooks) [inline .github/workflows/continuous-deployment.yml:256]: Do we need to have both of these jobs? `npm run coverage` runs all the tests to keep track of what is and is not covered. Failure output is similar.

File: .github/workflows/continuous-deployment.yml [hunk 2]
```diff
@@ -277,9 +284,15 @@ jobs:
...
 
       - name: Execute Tests
+        if: ${{ vars.ENVIRONMENT != 'Main-Gov' && github.event_name != 'pull_request' }}
         run: |
           npm run test
 
+      - name: Execute Coverage
+        if: ${{ vars.ENVIRONMENT == 'Main-Gov' || (github.event_name == 'pull_request' && github.base_ref == 'main') }}
+        run: |
+          npm run coverage
```
Comment (jamesobrooks) [inline .github/workflows/continuous-deployment.yml:294]: Do we need both of these jobs?

**Review by jamesobrooks — COMMENTED** [review]
Not sure we need both `npm run test` and `npm run coverage` jobs.

**Review by jamesobrooks — APPROVED** [review]

**Pull request merged**